File: ios/Andrews/Model/Rewards.swift
import Foundation

enum Rewards {
    static func points(_ orders: [Order]) -> Int {
        let total = orders.reduce(Decimal.zero) { $0 + $1.total }
        var rounded = Decimal.zero
        var value = total
        NSDecimalRound(&rounded, &value, 0, .down)
        return NSDecimalNumber(decimal: rounded).intValue
    }

    static func tier(_ points: Int) -> (name: String, nextName: String?, nextAt: Int?) {
        if points >= 1_500 {
            return ("Gold", nil, nil)
        }
        if points >= 500 {
            return ("Cardinal", "Gold", 1_500)
        }
        return ("Cyclone", "Cardinal", 500)
    }
}

File: src/lib/rewards.ts
import type { Order } from "../data/types";

type Tier = {
  name: "Cyclone" | "Cardinal" | "Gold";
  nextName: string | null;
  nextAt: number | null;
};

export function points(orders: Order[]): number {
  return Math.floor(orders.reduce((sum, order) => sum + order.total, 0));
}

export function tier(pts: number): Tier {
  if (pts >= 1500) return { name: "Gold", nextName: null, nextAt: null };
  if (pts >= 500) return { name: "Cardinal", nextName: "Gold", nextAt: 1500 };
  return { name: "Cyclone", nextName: "Cardinal", nextAt: 500 };
}
